There are several warnings from golint during build

golint generates multiple warnings during build:

```
pkg/chartutil/chartfile.go:34:7:warning: const ApiVersionV1 should be APIVersionV1 (golint)
pkg/chartutil/files.go:214:6:warning: func ToJson should be ToJSON (golint)
pkg/chartutil/files.go:229:6:warning: func FromJson should be FromJSON (golint)
pkg/getter/httpgetter.go:30:1:warning: comment on exported type HttpGetter should be of the form "HttpGetter ..." (with optional leading article) (golint)
pkg/getter/httpgetter.go:31:6:warning: type HttpGetter should be HTTPGetter (golint)
pkg/getter/httpgetter.go:37:1:warning: exported method HttpGetter.SetCredentials should have comment or be unexported (golint)
pkg/repo/chartrepo.go:199:1:warning: comment on exported function FindChartInAuthRepoURL should be of the form "FindChartInAuthRepoURL ..." (golint)
pkg/repo/repo.go:34:6:warning: type name will be used as repo.RepoFile by other packages, and that stutters; consider calling this File (golint)
pkg/strvals/parser.go:53:1:warning: comment on exported function ParseString should be of the form "ParseString ..." (golint)
```

Two of the above (`ToJson` and `FromJson`) are already being addressed by PR #3753 .

--- a/pkg/chartutil/chartfile.go
+++ b/pkg/chartutil/chartfile.go
@@ -31,7 +31,7 @@ import (
 // ApiVersionV1 is the API version number for version 1.
 //
 // This is ApiVersionV1 instead of APIVersionV1 to match the protobuf-generated name.
-const ApiVersionV1 = "v1"
+const ApiVersionV1 = "v1" // nolint
 
 // UnmarshalChartfile takes raw Chart.yaml data and unmarshals it.
 func UnmarshalChartfile(data []byte) (*chart.Metadata, error) {

--- a/pkg/downloader/chart_downloader.go
+++ b/pkg/downloader/chart_downloader.go
@@ -151,7 +151,7 @@ func (c *ChartDownloader) ResolveChartVersion(ref, version string) (*url.URL, ge
 	return u, nil, err
 }
 
-// Same as the ResolveChartVersion method, but returns the chart repositoryy.
+// ResolveChartVersionAndGetRepo is the same as the ResolveChartVersion method, but returns the chart repositoryy.
 func (c *ChartDownloader) ResolveChartVersionAndGetRepo(ref, version string) (*url.URL, *repo.ChartRepository, *getter.HttpGetter, error) {
 	u, err := url.Parse(ref)
 	if err != nil {

--- a/pkg/getter/httpgetter.go
+++ b/pkg/getter/httpgetter.go
@@ -27,13 +27,15 @@ import (
 	"k8s.io/helm/pkg/version"
 )
 
-//httpGetter is the efault HTTP(/S) backend handler
-type HttpGetter struct {
+//HttpGetter is the efault HTTP(/S) backend handler
+// TODO: change the name to HTTPGetter in Helm 3
+type HttpGetter struct { //nolint
 	client   *http.Client
 	username string
 	password string
 }
 
+//SetCredentials sets the credentials for the getter
 func (g *HttpGetter) SetCredentials(username, password string) {
 	g.username = username
 	g.password = password

--- a/pkg/repo/chartrepo.go
+++ b/pkg/repo/chartrepo.go
@@ -196,9 +196,9 @@ func FindChartInRepoURL(repoURL, chartName, chartVersion, certFile, keyFile, caF
 	return FindChartInAuthRepoURL(repoURL, "", "", chartName, chartVersion, certFile, keyFile, caFile, getters)
 }
 
-// FindChartInRepoURL finds chart in chart repository pointed by repoURL
-// without adding repo to repositories.
-// Unlike the FindChartInRepoURL function, this function also receives credentials for the chart repository.
+// FindChartInAuthRepoURL finds chart in chart repository pointed by repoURL
+// without adding repo to repositories, like FindChartInRepoURL,
+// but it also receives credentials for the chart repository.
 func FindChartInAuthRepoURL(repoURL, username, password, chartName, chartVersion, certFile, keyFile, caFile string, getters getter.Providers) (string, error) {
 
 	// Download and write the index file to a temporary location

--- a/pkg/repo/repo.go
+++ b/pkg/repo/repo.go
@@ -31,7 +31,8 @@ import (
 var ErrRepoOutOfDate = errors.New("repository file is out of date")
 
 // RepoFile represents the repositories.yaml file in $HELM_HOME
-type RepoFile struct {
+// TODO: change type name to File in Helm 3 to resolve linter warning
+type RepoFile struct { // nolint
 	APIVersion   string    `json:"apiVersion"`
 	Generated    time.Time `json:"generated"`
 	Repositories []*Entry  `json:"repositories"`

--- a/pkg/strvals/parser.go
+++ b/pkg/strvals/parser.go
@@ -50,7 +50,7 @@ func Parse(s string) (map[string]interface{}, error) {
 	return vals, err
 }
 
-// Parse parses a set line and forces a string value.
+// ParseString parses a set line and forces a string value.
 //
 // A set line is of the form name1=value1,name2=value2
 func ParseString(s string) (map[string]interface{}, error) {

